File: src/main/java/de/fxnm/ui/errormessage/ErrorMessage.java
package de.fxnm.ui.errormessage;

import javax.swing.Icon;
import javax.swing.JComponent;

public class ErrorMessage extends JComponent {

    private final String errorMessage;
    private final Icon icon;

    public ErrorMessage(final Icon icon, final String errorMessage) {
        this.icon = icon;
        this.errorMessage = errorMessage;
    }


    public Icon getIcon() {
        return this.icon;
    }

    @Override
    public String toString() {
        return this.errorMessage;
    }
}

File: src/main/java/de/fxnm/ui/errormessage/ErrorMessage.kt
package de.fxnm.ui.errormessage

import java.awt.BorderLayout
import javax.swing.Icon
import javax.swing.JLabel
import javax.swing.JPanel

class ErrorMessage(private val icon: Icon, private val message: String, val autoRemove: Boolean) : JPanel(BorderLayout()) {
    init {
        this.add(JLabel(icon), BorderLayout.WEST)
        this.add(JLabel(message), BorderLayout.CENTER)
    }
}
